trieste/models/gpflux/models.py

```diff
@@ -0,0 +1,112 @@
+# Copyright 2021 The Trieste Contributors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from typing import Optional
+
+import gpflow
+import gpflux
+import tensorflow as tf
+import tensorflow_probability as tfp
+from gpflux.models import DeepGP
+from gpflux.models.deep_gp import sample_dgp
+from gpflux.layers import GPLayer
+import warnings
+
+from ...data import Dataset
+from ...types import TensorType
+from ...utils import jit
+from ..interfaces import TrainableProbabilisticModel
+from ..optimizer import TFOptimizer
+from .interface import GPfluxPredictor
+
+
+class VanillaDeepGP(GPfluxPredictor, TrainableProbabilisticModel):
+    """
+    A :class:`TrainableProbabilisticModel` wrapper for a GPflux :class:`~gpflow.models.DeepGP` with
+    only standard :class:`GPLayer`\s: this class does not support keras layers, latent variable
+    layers, etc.
+    """
+
+    def __init__(
+        self, model: DeepGP, optimizer: TFOptimizer | None = None
```

Comment:
if creating a `DeepGP` model is a bit more involved than for the usual `GPflow`, perhaps its worthwhile providing one or two constructors for popular variants? if so, I would do it in a separate file

Reply:
Yes, I was thinking this, perhaps I could make an `architectures` file or folder in the `gpflux` folder?

Reply:
a single file should do the trick, if you need some supporting functions, put them in a `util.py` file

Reply:
I've had a go at this for a simple dgp architecture, in `trieste\models\gpflux\architectures.py`